specification/nginx/resource-manager/NGINX.NGINXPLUS/preview/2025-03-01-preview/swagger.json

```diff
@@ -0,0 +1,2850 @@
+      "enum": [
+        "NotStarted",
+        "InProgress",
+        "Succeeded",
+        "Failed"
+      ],
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "NginxDeploymentWafPolicyCompilingStatusCode",
+        "values": [
+          {
+            "value": "NotStarted",
+            "description": "The compilation of the custom waf policy has not started"
+          },
+          {
+            "value": "InProgress",
+            "description": "The compilation of the custom waf policy is in progress"
+          },
+          {
+            "value": "Succeeded",
+            "description": "The compilation of the custom waf policy is completed successfully and can now be referenced in the nginx config."
+          },
+          {
+            "value": "Failed",
+            "description": "The compilation of the custom waf policy failed."
+          }
+        ]
+      }
+    },
+    "NginxDeploymentDefaultWafPolicyListResponse": {
+      "type": "object",
+      "title": "Nginx Deployment Default Waf Policy List Response",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NginxDeploymentDefaultWafPolicyProperties"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "type": "string"
+        }
+      }
+    },
+    "NginxDeploymentWafPolicyListResponse": {
+      "type": "object",
+      "title": "Nginx Deployment Waf Policy List Response",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NginxDeploymentWafPolicyMetadata"
+          }
+        },
+        "nextLink": {
+          "type": "string"
+        }
+      }
+    },
+    "NginxDeploymentWafPolicyProperties": {
+      "type": "object",
+      "title": "Nginx Deployment Waf Policy Properties",
+      "properties": {
+        "provisioningState": {
+          "$ref": "#/definitions/ProvisioningState"
+        },
+        "content": {
+          "format": "byte",
+          "type": "string",
+          "description": "The byte content of the Policy"
+        },
+        "filepath": {
+          "type": "string",
+          "description": "The file path where the Policy is to be saved"
+        },
+        "compilingState": {
+          "$ref": "#/definitions/NginxDeploymentWafPolicyCompilingStatus"
+        },
+        "applyingState": {
+          "$ref": "#/definitions/NginxDeploymentWafPolicyApplyingStatus"
+        }
+      }
+    },
+    "NginxDeploymentWafPolicyMetadataProperties": {
+      "type": "object",
+      "title": "Nginx Deployment Waf Policy Metadata Properties",
+      "properties": {
+        "filepath": {
+          "type": "string",
+          "readOnly": true
+        },
+        "provisioningState": {
+          "$ref": "#/definitions/ProvisioningState"
```

Comment:
>ProvisioningState [](http://example.com/codeflow?start=33&length=17)

Do all your different resources really have the same provisioning state enum values? Just checking!!

Reply:
Yes that's the case